oauth2/runtime-flink-tests/build.gradle.kts

```
@@ -23,72 +23,188 @@ description = "Flink tests for Dremio AuthManager for Apache Iceberg"
+        dependsOn(":authmgr-oauth2-runtime:shadowJar")
 
-  intTestImplementation(libs.assertj.core)
-  intTestImplementation(libs.mockito.core)
+        environment("AWS_REGION", "us-west-2")
+        environment("AWS_ACCESS_KEY_ID", "fake")
+        environment("AWS_SECRET_ACCESS_KEY", "fake")
+
+        jvmArgs(
+          "--add-exports",
+          "java.base/sun.nio.ch=ALL-UNNAMED",
+          "--add-opens",
+          "java.base/java.util=ALL-UNNAMED",
+          "--add-opens",
+          "java.base/java.lang=ALL-UNNAMED",
+          "--add-opens",
+          "java.base/java.lang.reflect=ALL-UNNAMED",
+          "--add-opens",
+          "java.base/java.io=ALL-UNNAMED",
+          "--add-opens",
+          "java.base/java.net=ALL-UNNAMED",
+          "--add-opens",
+          "java.base/java.nio=ALL-UNNAMED",
+          "--add-opens",
+          "java.base/java.util.concurrent=ALL-UNNAMED",
+          "--add-opens",
+          "java.base/java.security=ALL-UNNAMED",
+        )
+
+        // Set system properties to identify the versions being tested
+        systemProperty("authmgr.test.iceberg.version", icebergVersion)
+        systemProperty("authmgr.test.flink.version", flinkVersion)
+
+        // Use unique working directory and reports
+        workingDir = layout.buildDirectory.dir("tmp/$taskName").get().asFile
+
+        // Use test name suffix to distinguish between matrix runs
+        reports {
+          junitXml.required.set(true)
+          junitXml.outputLocation.set(layout.buildDirectory.dir("test-results/$taskName"))
+          html.required.set(true)
+          html.outputLocation.set(layout.buildDirectory.dir("reports/tests/$taskName"))
+        }
+
+        // Ensure unique inputs to avoid caching conflicts
+        inputs.property("icebergVersion", icebergVersion)
+        inputs.property("flinkVersion", flinkVersion)
+      }
+
+    matrixTestTasks.add(testTask)
+  }
 }
 
 tasks.named<Test>("intTest").configure {
-  if (System.getenv("CI") == null) {
-    maxParallelForks = 2
-  }
-  dependsOn(":authmgr-oauth2-runtime:shadowJar")
-  environment("AWS_REGION", "us-west-2")
-  environment("AWS_ACCESS_KEY_ID", "fake")
-  environment("AWS_SECRET_ACCESS_KEY", "fake")
-  jvmArgs(
-    "--add-exports",
-    "java.base/sun.nio.ch=ALL-UNNAMED",
-    "--add-opens",
-    "java.base/java.util=ALL-UNNAMED",
-    "--add-opens",
-    "java.base/java.lang=ALL-UNNAMED",
-    "--add-opens",
-    "java.base/java.lang.reflect=ALL-UNNAMED",
-    "--add-opens",
-    "java.base/java.io=ALL-UNNAMED",
-    "--add-opens",
-    "java.base/java.net=ALL-UNNAMED",
-    "--add-opens",
-    "java.base/java.nio=ALL-UNNAMED",
-    "--add-opens",
-    "java.base/java.util.concurrent=ALL-UNNAMED",
-    "--add-opens",
-    "java.base/java.security=ALL-UNNAMED",
+  dependsOn(
+    tasks.named(
+      "intTest_iceberg${defaultIcebergVersion.replace(".", "")}_flink${defaultFlinkVersion.replace(".", "")}"
```

NIt: better replace with `_` (1.11.1 would be 1.1.11)
```suggestion
      "intTest_iceberg${defaultIcebergVersion.replace(".", "_")}_flink${defaultFlinkVersion.replace(".", "_")}"
```